### PR TITLE
Dealer frontend fixes

### LINKED ISSUE #1282: [MAGClassic] Vendor special requests and address not being addedto db
After submitting the dealer application form, the special request and address fields are not shown on the corresponding group page.

### LINKED ISSUE #1283: [Classic] Add dealer table and badge prices
Add a note on the dealer applicaiton form below the "tables" dropdown indicating that they are $80 per table, and under the Badges indicating they are $20 per badge.

I tried to fight back on those prices, but didn't have the energy.

### LINKED ISSUE #1286: [Classic] Error when paying for group.
Tried to pay for this group:
https://stage.uber.magfest.org/uber/groups/form?id=b32b4e3d-33bc-43ee-bbc8-35cd6a00e139&message=Group%20info%20uploaded

On this page:
https://stage.uber.magfest.org/uber/preregistration/group_members?id=b32b4e3d-33bc-43ee-bbc8-35cd6a00e139

Upon entering credit card info and attempting to proceed, got this message:

Traceback (most recent call last): File "/usr/local/uber/plugins/uber/uber/decorators.py", line 117, in charge return func(self, session=session, payment_id=payment_id, stripeToken=stripeToken) File "/usr/local/uber/plugins/uber/uber/site_sections/preregistration.py", line 358, in process_group_payment render('emails/dealers/payment_notification.txt', {'group': group}), model=group) File "/usr/local/uber/plugins/uber/uber/decorators.py", line 193, in render template = loader.select_template(listify(template_name_list)) File "/usr/local/uber/env/lib/python3.4/site-packages/django/template/loader.py", line 186, in select_template raise TemplateDoesNotExist(', '.join(not_found)) django.template.base.TemplateDoesNotExist: emails/dealers/payment_notification.txt

### LINKED ISSUE #1287: [Classic] Deadlines on group member kickins
Attempting to edit this group badge:
https://stage.uber.magfest.org/uber/preregistration/register_group_member?group_id=b32b4e3d-33bc-43ee-bbc8-35cd6a00e139

Registration deadlines should be shown on the badge level buttons.

### LINKED ISSUE #1288: Change dealer badge form to max out at 8
When applying as a dealer, attendees should only be able to select a maximum of 8 badges.

### PR DESCRIPTION
**Pull request by kitsuta**
Miscellaneous fixes to the workflow and UI for dealer registration.

Fixes #1287 
Fixes #1286 
Fixes #1283 
Fixes #1288 
Fixes #1282

### CODE REVIEW

**Comment by earl7399** [comment]
These look good to my (admittedly inexperienced) eye.

File: uber/templates/groups/form.html [hunk 1]
```diff
@@ -183,13 +183,57 @@
             <textarea rows="3" cols="30" name="special_needs">{{ group.special_needs }}</textarea>
         </div>
     </div>
```
Comment (EliAndrewC) [inline uber/templates/groups/form.html]: This suppresses these form fields in cases where the leader doesn't already have an address filled out.  Won't that make it impossible for an admin to enter this information if it's not already entered?

Comment (kitsuta) [inline uber/templates/groups/form.html]: It gives a warning to the admin - I haven't tested this explicitly, but you
should be able to go into the group leader's details and edit their address
there.

**Comment by EliAndrewC** [comment]
In general I get the motivation using the attendee for address; since we already have fields on the attendee table for address, it makes sense to use those fields instead of having a separate address box on the group table.

However, I can forsee a lot of problems and edge cases.  Off the top of my head:
- an admin can change the group leader to someone else, meaning that the dealer would suddenly no longer have an address
- the leader could transfer their badge to someone else, which would unset the address
- a group could just not have a leader in the first place, which means we can't have an address for the dealer until it does
- by using the attendee address, there's no way for the attendee's address and the business' address to be different (admittedly, the attendee address isn't currently used, but this would be a problem if it was)

We could theoretically address these edge cases, but it seems like that would add far more complexity to the code than simply using the existing address field on the group table.

Does all of that make sense?  I'm kind of inclined to reject this PR in favor of maintaining separate address tracking on the attendee and group tables.

**Comment by kitsuta** [comment]
Just FYI, the actual collection of the address has been in place since the
merge - this is merely a change to show it on the admin side.

I'm not opposed to going back to using the group's address field, but we'd
want to amend this PR to implement that rather than rejecting it.

**Comment by EliAndrewC** [comment]
> Just FYI, the actual collection of the address has been in place since the merge - this is merely a change to show it on the admin side.
> I'm not opposed to going back to using the group's address field, but we'd want to amend this PR to implement that rather than rejecting it.

Gotcha, yeah I think we should do that.  I'm working with Brent tonight on some Stops stuff, but I'll see if I can get to that before I go to bed.

File: uber/models.py [hunk 2]
```diff
@@ -1180,10 +1180,6 @@ def is_dept_head(self):
         return self.ribbon == c.DEPT_HEAD_RIBBON
 
     @property
-    def is_group_leader(self):
```
Comment (kitsuta) [inline uber/models.py]: Unrelated change - there's an exact copy of this function a few lines down.

**Comment by kitsuta** [comment]
I went ahead and made the change. :)

**Comment by binary1230** [comment]
+1 do et

**Comment by EliAndrewC** [comment]
Great stuff!
